pkg/cluster/internal/providers/docker/network.go

```diff
@@ -102,6 +130,55 @@ func createNetwork(name, ipv6Subnet string) error {
 		"--ipv6", "--subnet", ipv6Subnet, name).Run()
 }
 
+func sortedNetworksWithName(name string) ([]string, error) {
+	// list all networks by this name
+	out, err := exec.Output(exec.Command(
+		"docker", "network", "ls",
+		"--filter=name=^"+regexp.QuoteMeta(name)+"$",
+		"--format={{json .}}",
+	))
+	if err != nil {
+		return nil, err
+	}
+
+	// parse
+	type networkLSEntry struct {
+		CreatedAt goDefaultTime `json:"CreatedAt"`
+		ID        string        `json:"ID"`
+	}
+
+	networks := []networkLSEntry{}
+	decoder := json.NewDecoder(bytes.NewReader(out))
+	for {
+		var network networkLSEntry
+		err := decoder.Decode(&network)
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, errors.Wrap(err, "failed to decode networks list")
+		}
+		networks = append(networks, network)
+	}
+
+	// deterministically sort networks
+	// NOTE: THIS PART IS IMPORTANT!
+	// TODO(fixme): we should be sorting on active usage first!
```

Comment:
this TODO is the one thing keeping this WIP.
I will come back and rework this later. the overall code will be more or less the same otherwise.